src/util/strencodings.cpp

```diff
@@ -526,3 +527,45 @@ std::string HexStr(const Span<const uint8_t> s)
     assert(it == rv.end());
     return rv;
 }
+
+std::optional<uint64_t> ParseByteUnits(const std::string& str, const uint64_t default_multiplier)
+{
+    auto multiplier = default_multiplier ? default_multiplier : 1;
+    char unit = str.empty() ? 0 : str.back();
+    switch (unit) {
+        case 'k':
+            multiplier = 1000;
+            break;
+        case 'K':
+            multiplier = 1024;
+            break;
+        case 'm':
+            multiplier = 1000 * 1000;
+            break;
+        case 'M':
+            multiplier = 1ULL << 20;
+            break;
+        case 'g':
+            multiplier = 1000ULL * 1000 * 1000;
+            break;
+        case 'G':
+            multiplier = 1ULL << 30;
+            break;
+        case 't':
+            multiplier = 1000ULL * 1000 * 1000 * 1000;
+            break;
+        case 'T':
+            multiplier = 1ULL << 40;
+            break;
+        default:
+            unit = 0;
+            break;
+    }
+
+    uint64_t num;
+    if (!ParseUInt64(unit ? str.substr(0, str.size()-1) : str, &num) ||
```

Comment:
nit: missing spaces around `-`.

Reply:
When I added the space, it didn't look aesthetically pleasing to me, normally I do add a space, but this if statement looks better without 
I don't mind adding a space if you like or if someone else asks.

Reply:
The thing is that lots of people have lots of different preferences. "Aesthetically pleasant" for somebody may not be for another. The project uses `clang-format` to avoid endless low-value discussions about styling.

Reply:
@vasild I changed `const uint64_t`  to `uint64_t`